kura/org.eclipse.kura.deployment.agent/src/main/java/org/eclipse/kura/deployment/agent/impl/DeploymentAgent.java

```diff
@@ -466,16 +453,18 @@ private void removePackageFromConfFile(String packageName) {
         oldDeployedPackages.putAll(deployedPackages);
         deployedPackages.remove(packageName);
 
-        if (this.dpaConfPath == null) {
-            logger.warn("Configuration file not specified");
-            return;
+        if (!oldDeployedPackages.equals(deployedPackages)) {
+            writeDPAPropertiesFile(this.dpaConfPath, deployedPackages);
         }
+    }
 
-        if (oldDeployedPackages.equals(deployedPackages)) {
+    private static void writeDPAPropertiesFile(String dpaConfPath, Properties deployedPackages) {
```

Comment:
Why must the method be static?

Reply:
It doesn't. I decided to set it static since it doesn't depend on any field of the `DeploymentAgent` class. Now that you questioned it I realised that it doesn't make much sense since it is a private method... fixing.

Reply:
Wait, I wanted to stimulate a discussion on this refactoring. As it is it makes sense to be static:

- is not dependent on instance creation
- is an "utility" method
- no operation involves instance variables

I wanted to discuss the last point: why doesen't it use the instance variable? Is `this.dpaConfPath` variable even needed in the rest of the class? Does it make sense to have it as an instance variable?

Reply:
> why doesen't it use the instance variable? Is this.dpaConfPath variable even needed in the rest of the class? Does it make sense to have it as an instance variable?

The variable is needed as it represent the path of the `dpa.properties` file which is used extensively by the `DeploymentAgent`.

As for the `writeDPAPropertiesFile` method: I decided to pass it as a parameter because I wanted to be explicit about what the method depends on.

Reply:
It is used extensively but unecessarly, in my opinion. The only place besides the methods touched so far by this PR is in `readDeployedPackages`. The read method is unecessarly invoked many times, can't we read the packages at activation and maintain the current properties as an instance variable? It would be much more efficient and readable.

Reply:
> The read method is unecessarly invoked many times, can't we read the packages at activation and maintain the current properties as an instance variable and update just that? It would be much more efficient and readable.

Arguably the efficiency here is not that important: how many times in the lifetime of a Kura device we see a deployment package installation? I'll take the read from disk over increasing memory usage for storing and maintaining the package list.

I'll leave it as-is.

Reply:
Agree with that. At this point I would remove the parameter from your write method and use the instance variable.